### DISCUSSION TITLE
How to access resolver members in subscription handler/filter?

### DISCUSSION THREAD

**Discussion by oliversalzburg** [MichalLytek/type-graphql#949] [Q&A] [answered]
When handling an attempt of a user to establish a subscription, I only have the `subscribe` callback to work with. This callback isn't bound to the resolver, so I don't have access to a lot of information associated with the resolver.

I want to validate access permissions for users attempting to establish a subscription on entities they aren't allowed to access. For that I would like to use the database access that is already established in the resolver, but I don't see how that would be possible.

The filter, provided through `withFilter` isn't bound to the resolver either, making it just as limited.

How should this usually be approached? I'm assuming I'm missing a different approach that is far better suited.

**Comment by MichalLytek** [comment, marked as answer]
If you use dependency injection, you can call `Container.get()` to get the instance of a service you use in resolver and call some methods there.

**Reply by oliversalzburg** [reply]
Fair enough. I was hoping I could just use the instances of these services that I already have established in the resolver.

Knowing that I haven't missed obvious alternatives is very helpful to know regardless. Thanks a lot.

**Reply by MichalLytek** [reply]
There is a proposal for a new API #542 but nothing is implemented yet.

**Reply by oliversalzburg** [reply]
Heh. Seems like I've crossed paths with that ticket in the past. Thanks for pointing it out.
